src/commitment/kzg.rs

```diff
@@ -0,0 +1,234 @@
+/// defined without depending on pairings.
+use ark_ec::{pairing::Pairing, CurveGroup, VariableBaseMSM};
+use ark_ff::PrimeField;
+use ark_poly::{
+    univariate::{DenseOrSparsePolynomial, DensePolynomial},
+    DenseUVPolynomial, EvaluationDomain, Evaluations, GeneralEvaluationDomain, Polynomial,
+};
+use ark_poly_commit::kzg10::{VerifierKey, KZG10};
+use ark_std::rand::Rng;
+use ark_std::{borrow::Cow, fmt::Debug};
+use ark_std::{One, Zero};
+use core::marker::PhantomData;
+use rayon::iter::{IntoParallelRefIterator, ParallelIterator};
+
+use super::CommitmentProver;
+use crate::transcript::Transcript;
+use crate::Error;
+
+/// ProverKey defines a similar struct as in ark_poly_commit::kzg10::Powers, but instead of
+/// depending on the Pairing trait it depends on the CurveGroup trait.
+#[derive(Debug, Clone, Default, Eq, PartialEq)]
+pub struct ProverKey<'a, C: CurveGroup> {
+    /// Group elements of the form `β^i G`, for different values of `i`.
+    pub powers_of_g: Cow<'a, [C::Affine]>,
+}
+
+pub struct KZGSetup<P: Pairing> {
+    _p: PhantomData<P>,
+}
+impl<'a, P> KZGSetup<P>
+where
+    P: Pairing,
+{
+    /// setup returns the tuple (ProverKey, VerifierKey). For real world deployments the setup must
+    /// be computed in the most trustless way possible, usually through a MPC ceremony.
+    pub fn setup<R: Rng>(rng: &mut R, len: usize) -> (ProverKey<'a, P::G1>, VerifierKey<P>) {
+        let len = len.next_power_of_two();
+        let universal_params = KZG10::<P, DensePolynomial<P::ScalarField>>::setup(len, false, rng)
+            .expect("Setup failed");
+        let powers_of_g = universal_params.powers_of_g[..=len].to_vec();
+        let powers = ProverKey::<P::G1> {
+            powers_of_g: ark_std::borrow::Cow::Owned(powers_of_g),
+        };
+        let vk = VerifierKey {
+            g: universal_params.powers_of_g[0],
+            gamma_g: universal_params.powers_of_gamma_g[&0],
+            h: universal_params.h,
+            beta_h: universal_params.beta_h,
+            prepared_h: universal_params.prepared_h.clone(),
+            prepared_beta_h: universal_params.prepared_beta_h.clone(),
+        };
+        (powers, vk)
+    }
+}
+
+/// KZGProver implements the CommitmentProver trait for the KZG commitment scheme.
+pub struct KZGProver<'a, C: CurveGroup> {
+    _a: PhantomData<&'a ()>,
+    _c: PhantomData<C>,
+}
+impl<'a, C> CommitmentProver<C> for KZGProver<'a, C>
+where
+    C: CurveGroup,
+{
+    type Params = ProverKey<'a, C>;
+    /// Proof is a tuple containing (evaluation, proof)
+    type Proof = (C::ScalarField, C);
+
+    /// commit implements the CommitmentProver commit interface, adapting the implementation from
+    /// https://github.com/arkworks-rs/poly-commit/tree/c724fa666e935bbba8db5a1421603bab542e15ab/poly-commit/src/kzg10/mod.rs#L178
+    /// with the main difference being the removal of the blinding factors and the no-dependancy to
+    /// the Pairing trait.
```

Comment:
We might want to credit them in the README (the WIP branch).

Reply:
Agree. And not only for the KZG10, but for all the other arkworks crates that we use ^^